### PR TITLE
Implemented readonly after having selected an entry

### PR DESCRIPTION
**Pull request by christophscheuing**
see issue: #1512

The additional "x" in the input field now allows to unselect a previously chosen entry from the autocomplete list. The requested user confirmation has been adapted to the two possible cases (having edited a new entity and then selecting one from the autocomplet list or having changed the values of an entity selected from the autocomplete list and then discarding this entry).

### CODE REVIEW

**Review by sleidig — APPROVED** [review]
Good job! This makes the UI more fool proof and clear :+1:

File: src/app/core/entity-components/entity-utils/dynamic-form-components/edit-text-with-autocomplete/edit-text-with-autocomplete.component.html [hunk 1]
```diff
@@ -33,12 +34,21 @@
...
   class="tooltip-suffix"
 ></fa-icon>
 
-
+<fa-icon
+  *ngIf="!autocompleteDisabled && !!selectedEntity"
+  (click)="resetForm()"
+  #tooltipElement="matTooltip"
+  icon="circle-xmark"
+  [matTooltip]="'Remove link to this existing record.'"
```
Comment (sleidig) [inline src/app/core/entity-components/entity-utils/dynamic-form-components/edit-text-with-autocomplete/edit-text-with-autocomplete.component.html:52]:
```suggestion
  matTooltip="Unload existing record and reset form."
```

**Comment by TheSlimvReal** [comment]
The reset does not work correctly:

1. Click the plus button
2. Select a autocomplete activity
3. Remove the autocomplete entity
4. Set a **new** name (don't select a another existing activity)
5. Save it
6. Go to the recurring activities and look for the previously "created" one
7. Open the details page
8. There you see that other students are already linked to this activity

Even after the `reset()` code is executed, the component still overwrites the previously selected entity.
This is because the hidden form fields like `_id` that are added in the `setAllFormValues` are **not** reset. In the reset function, these should even be removed from the form again.

**Review by TheSlimvReal — CHANGES_REQUESTED** [review]
Fix error described in the above comment (with a (improved) test, that ensures that this error is actually fixed)

**Comment by christophscheuing** [comment]
I've tried and fixed the described error. Maybe there's a more elegant way than using an extra class variable for the added controls.

**Review by TheSlimvReal — APPROVED** [review]
Nice works well now. But the component is getting a bit overly complicated. Lets hope we wont have to touch this too often

**Pull request merged**